Add the PM update for May 2026 (#1868)
rust-lang/blog.rust-lang.org: content/inside-rust/program-management-update-2026-05.md

+++
path = "inside-rust/2026/06/11/program-management-update-may-2026"
title = "Program management update — May 2026"
authors = ["Tomáš Šedovič and Nurzhan Saken"]

[extra]
team = "the Program team"

> team_url = "https://www.rust-lang.org/governance/teams/launching-pad#team-program"

Reviewer: I don't remember if I've mentioned this before, but the `#team-program` part of this link doesn't resolve to anything. Unfortunately marker teams are not on the website.

*[View changes since the review](https://triagebot.infra.rust-lang.org/gh-changes-since/rust-lang/blog.rust-lang.org/1868/7bf52ffa44ade9b713f61b7ce7705c39ceaefb76..fa8af786da4ac4f55342f015fe6379afbbbded1c)*

Reviewer 2: They can be added to the website, it's enough to add a [website] entry for them in the team repo.

Author: Yeah I noticed that too. Would you prefer we not use the URL next time? It's autogenerated from the CLI (which does have a program team listed).

Reviewer 2: Let's do something better: add you to the website :) https://github.com/rust-lang/team/pull/2515

+++
# Program management update — May 2026

## RustWeek and All Hands!

> It finally happened: Nurzhan, Tomáš, and hundreds of Rustaceans all met up in Utrecht, the Netherlands, for [RustWeek and All Hands](https://2026.rustweek.org/).

Reviewer 3: ```suggestion
It finally happened. Nurzhan, Tomáš, and hundreds of Rustaceans all met up in Utrecht, the Netherlands, for [RustWeek and All Hands](https://2026.rustweek.org/).
```

> RustWeek is a week-long conference organized by [RustNL](https://rustnl.org/). It typically features two days of talks, three to four days of workshops, an unconference, and a hackathon. It's absolutely fantastic and open to everyone. The All Hands is a three day event of team- and topic-specific sessions, bringing Project members together to meet, work together, resolve issues, and plan ahead. For the last two years, these events were co-located, so most Project members got to attend both.

Reviewer 4: ```suggestion
RustWeek is a week-long conference organized by [RustNL](https://rustnl.org/). It typically features two days of talks, three to four days of workshops, an unconference, and a hackathon. It's absolutely fantastic and open to everyone. The All Hands is a three-day event of team- and topic-specific sessions, bringing Project members together to meet, collaborate, resolve issues, and plan ahead. For the last two years, these events were co-located, so most Project members got to attend both.
```

> RustWeek is a week-long conference organized by [RustNL](https://rustnl.org/). It typically features two days of talks, three to four days of workshops, an unconference, and a hackathon. It's absolutely fantastic and open to everyone. The All Hands is a three-day event of team- and topic-specific sessions, bringing Project members together to meet, collaborate, resolve issues, and plan ahead. For the last two years, these events have been co-located, so most Project members have been able to attend both.

Reviewer 3: ```suggestion
RustWeek is a week-long conference organized by [RustNL](https://rustnl.org/). It typically features two days of talks, three to four days of workshops, an unconference, and a hackathon. It's absolutely fantastic and open to everyone. The All Hands is a three-day event of team- and topic-specific sessions, bringing Project members together to meet, collaborate, resolve issues, and plan ahead. For the last two years, these events have been co-located, so most Project members got to attend both.
```

The dates for [RustWeek 2027](https://2027.rustweek.org/) are already set: May 24–29, in Utrecht again. The dates and location for the All Hands have not yet been announced.

We'll post a retrospective later, but in the meantime, it was fantastic to meet everyone.

### Interviews

During the conference, the Content team recorded 12 interviews with some of the speakers and people in the Project. We got involved, too: Nurzhan was really excited (and anxious!) to meet and interview Alice Cecile (Bevy Engine) about program and project management. The interviews have been fully produced by a videographer hired from the team's budget, and they should be posted in the next month or so.

> Apart from that, the team has a backlog of interviews recorded at the last Kangrejos and RustConf that is now more than nine months old. They hired an editor to handle them, so we're excited to see more interviews posted soon.

Reviewer 3: ```suggestion
Apart from that, the team has a backlog of interviews recorded at the last Kangrejos and RustConf that is over nine months old now. They hired an editor to take care of them, so we're excited to see more interviews posted soon.
```

## Program management

We published the [final update for the 2025H2 Project goals](https://blog.rust-lang.org/2026/05/18/project-goals-2026-04/), and kicked off the new 2026 period.

> We had some goals-related sessions at the All Hands and got to talk to people about their needs, and experiences. There was also a discussion about funding for maintainers which we plan to integrate into goals.

Reviewer 4: ```suggestion
We had some goals-related sessions at the All Hands and got to talk to people about their needs and experiences. There was also a discussion about funding for maintainers which we plan to integrate into goals.
```

We're now working on displaying funding information (goals looking for funding, goals that are funded, etc.) on the [Project Goals page](https://rust-lang.github.io/rust-project-goals/2026/index.html). For an example, see the [Funding section of the Fast Builds page](https://rust-lang.github.io/rust-project-goals/2026/roadmap-fast-builds.html#funding).

Some other things we're looking into:

* The feasibility of using goals ubiquitously instead of the current Compiler MCPs (Major Change Proposals), Library ACPs (API Change Proposals), and Language experiments.

>   * The Types team recently set a precedent by [deciding](https://github.com/rust-lang/types-team/issues/134) to move away from MCPs in favor of goals.

Reviewer 3: ```suggestion
  * The Types team set a precedent for this recently by [deciding](https://github.com/rust-lang/types-team/issues/134) to move away from MCPs in favor of goals.
```

* Making the creation of new goals more straightforward and lightweight
* Streamlining goal updates and making them more useful

> * Refactoring the tooling for managing goals

Reviewer 3: ```suggestion
* Making the creation of new goals more straightforward and lightweight.
* Streamlining goal updates and making them more useful.
* Refactoring the tooling for managing goals.
```

* Closer integration with the Funding team, grants, etc.

> Basically, we expect for goals to be opened throughout the year on a rolling basis. How do we handle that?

Reviewer 4: ```suggestion
Basically, we expect goals to be opened throughout the year on a rolling basis. How do we handle that?
```

> Basically, we expect goals to be opened throughout the year on a rolling basis. How should we handle that?

Reviewer 3: ```suggestion
Basically, we expect goals to be opened throughout the year on a rolling basis. How do we handle that?
```

We have also expanded the coverage of meetings. The Rustdoc team used to have chat-based meetings, but recently decided to try video calls instead, and asked us to minute them. We also reached out to the Types team asking them to move their meeting to a non-conflicting time slot (30 minutes earlier) so that we could start attending those too. Hopefully, this will help us keep the teams more connected.

> ## Funding Rust Maintainers

Reviewer 4: ```suggestion
## Funding Rust maintainers
```

The [Rust Foundation Maintainer Fund (RFMF) RFC](https://rust-lang.github.io/rfcs/3931-rfmf-rust-foundation-maintainer-fund.html) has been merged.

> The Foundation will begin raising money for the Maintainer Fund. This money will be dedicated to people doing maintenance work. This includes work such as reviewing, triaging, large-scale refactoring, and development that is largely invisible but critical to the Project's future or needed to unlock new features.

Reviewer 3: ```suggestion
The Foundation will begin raising money for the Maintainer Fund. This money will be dedicated to people doing maintenance work. This includes work such as reviewing, triaging, and large-scale refactoring, but also work that is largely invisible yet has far-reaching effects, e.g., unblocking new features.
```

> The Foundation will begin raising money for the Maintainer Fund. This money will be dedicated to people doing maintenance work. This includes things such as reviewing, triaging, large scale refactoring, and development that's largely invisible but critical to the Project going forward or needed to unlock new features.

Reviewer 4: ```suggestion
The Foundation will begin raising money for the Maintainer Fund. This money will be dedicated to people doing maintenance work. This includes things such as reviewing, triaging, large-scale refactoring, and development that's largely invisible but critical to the Project going forward or needed to unlock new features.
```

The RFC establishes a Maintainer in Residence program (MiR — not to be confused with [MIR](https://rustc-dev-guide.rust-lang.org/mir/index.html)) to provide long-term, mostly full-time support for maintainers — complementing the shorter-term [Grants program](https://github.com/rust-lang/rfcs/pull/3919).

This program was heavily inspired by Python's [Developers in Residence](https://www.python.org/psf/developersinresidence/).

> The [Funding team](https://rust-lang.org/governance/teams/launching-pad/#team-funding) wrote about the program [here](https://blog.rust-lang.org/2026/06/02/launching-the-rust-foundation-maintainers-fund/), and the Foundation published a [companion piece on its site](https://rustfoundation.org/media/help-fund-the-people-who-build-rust/).

Reviewer 3: ```suggestion
The [Funding team](https://rust-lang.org/governance/teams/launching-pad/#team-funding) wrote about the program [here](https://blog.rust-lang.org/2026/06/02/launching-the-rust-foundation-maintainers-fund/), and the Foundation published a [companion piece on their site](https://rustfoundation.org/media/help-fund-the-people-who-build-rust/).
```

> The [Funding team](https://rust-lang.org/governance/teams/launching-pad/#team-funding) wrote about [the program here](https://blog.rust-lang.org/2026/06/02/launching-the-rust-foundation-maintainers-fund/), and the Foundation published a [companion piece on their site](https://rustfoundation.org/media/help-fund-the-people-who-build-rust/).

Reviewer 4: ```suggestion
The [Funding team](https://rust-lang.org/governance/teams/launching-pad/#team-funding) wrote about the program [here](https://blog.rust-lang.org/2026/06/02/launching-the-rust-foundation-maintainers-fund/), and the Foundation published a [companion piece on their site](https://rustfoundation.org/media/help-fund-the-people-who-build-rust/).
```

If you're an individual, you can [donate money to the Maintainer Fund via this GitHub Sponsors page](https://github.com/sponsors/rustfoundation). If you're a company or larger organization, reach out to <maintainers-fund@rustfoundation.org>. All proceeds will go directly to maintainers.

If you want to support specific people, head to the [Funding page](https://rust-lang.org/funding/).

> And last but not least, RustNL has recently hired several Rust maintainers as well. [Visit its page to learn more](https://rustnl.org/maintainers/).

Reviewer 3: ```suggestion
And last but not least, RustNL has recently hired several Rust maintainers as well. [Visit their page to learn more](https://rustnl.org/maintainers/).
```

## Mirroring

At the All Hands, Joel Marcey distributed YubiKeys that will be used for mirror attestations. Emily, Walter, and Dirkjan put together a guide on how to use them (private for now, undergoing feedback) and verify attestation.

Arlo has practiced root signing and shared examples [here](https://github.com/arlosi/tuf-on-ci).

The team discussed whether [TUF](https://theupdateframework.io/) is the right solution for the crates.io index. TUF was originally designed for offline signing, but the crates.io index changes very frequently and will need to be signed every minute.

> The main concern was that following the TUF spec to the letter would be infeasible due to resulting in too many HTTP requests. The standard seems to be moving in the right direction to solve this (they're listening to our input), but we don't want be blocked on that.

Reviewer 4: ```suggestion
The main concern was that following the TUF spec to the letter would be infeasible due to resulting in too many HTTP requests. The standard seems to be moving in the right direction to solve this (they're listening to our input), but we don't want to be blocked on that.
```

> There are options the team could do, but they'd make crates.io incompatible with other TUF clients.

Reviewer 4: ```suggestion
There are options the team could consider, but they'd make crates.io incompatible with other TUF clients.
```

> Walter said that we don't necessarily need to support other clients for the crates.io usecase, and that the changes we'd need to do to [rust-tuf](https://github.com/theupdateframework/rust-tuf) aren't substantial enough for this to be a problem.

Reviewer 4: ```suggestion
Walter said that we don't necessarily need to support other clients for the crates.io use case, and that the changes we'd need to do to [rust-tuf](https://github.com/theupdateframework/rust-tuf) aren't substantial enough for this to be a problem.
```

Rustup doesn't have the same constraints, so we'll be following the spec there.

## Rust for Linux

### All Hands session

The Rust for Linux team was invited to the All Hands and hosted an office hours session there. There were rustfmt and clippy representatives in [the room](https://github.com/rust-lang/all-hands-2026/issues/18).

> Jieyou Xu wanted to clarify what exactly the Linux team needs in terms of removing the ["trailing double slash hack"](https://docs.kernel.org/rust/coding-guidelines.html#imports). The team wants this option to be stabilized at some point, but in the meantime they'd be okay with using something like `RUSTC_BOOTSTRAP` (which allows using unstable features with a stable toolchain). Rustfmt, however, doesn't have a way to do this today.

Reviewer 4: ```suggestion
Jieyou Xu wanted to clarify what exactly the Linux team needs to remove the ["trailing double slash hack"](https://docs.kernel.org/rust/coding-guidelines.html#imports). The team wants this option to be stabilized at some point, but in the meantime they'd be okay with using something like `RUSTC_BOOTSTRAP` (which allows using unstable features with a stable toolchain). Rustfmt, however, doesn't have a way to do this today.
```

> Jieyou Xu wanted to clarify exactly what the Linux team needs to remove the ["trailing double slash hack"](https://docs.kernel.org/rust/coding-guidelines.html#imports). The team wants this option to be stabilized at some point, but in the meantime, they would be okay with using something like `RUSTC_BOOTSTRAP` (which allows unstable features to be used with a stable toolchain). Rustfmt, however, does not have a way to do this today.

Reviewer 3: ```suggestion
Jieyou Xu wanted to clarify what exactly the Linux team needs to remove the ["trailing double slash hack"](https://docs.kernel.org/rust/coding-guidelines.html#imports). The team wants this option to be stabilized at some point, but in the meantime, they would be okay with using something like `RUSTC_BOOTSTRAP` (which allows unstable features to be used with a stable toolchain). Rustfmt, however, does not have a way to do this today.
```

> The team runs rustfmt in CI, but they're currently only looking for crashes (ICE — internal compiler error). They want get to a place where they could catch unintentional differences in code formatting, and to be able to enforce consistent style in CI.

Reviewer 4: ```suggestion
The team runs rustfmt in CI, but they're currently only looking for crashes (ICE — internal compiler error). They want get to a place where they could catch unintentional differences in code formatting, and enforce consistent style in CI.
```

> Alejandra González mentioned that getting changes into clippy are now bottle-necked by reviewers. Reviewing rustfmt code is complex. There's a lot of things to check, and edge-cases to investigate. But Alejandra stressed that if someone is interested, they'd be happy to teach them, and they'd appreciate the help.

Reviewer 5: is it intentional that this first references clippy and then rustfmt?

*[View changes since the review](https://triagebot.infra.rust-lang.org/gh-changes-since/rust-lang/blog.rust-lang.org/1868/7bf52ffa44ade9b713f61b7ce7705c39ceaefb76..29d15e9fda512dbddc608f4311a8585c551a16df)*

Author: Oops, that's a typo. Thanks for spotting!

> Alejandra González mentioned that getting changes into clippy are now bottle-necked by reviewers. Reviewing rustfmt code is complex. There's a lot of things to check, and edge-cases to investigate. But Alejandra stressed that if someone is interested, they'd be happy to teach them, and they'd appreciate the help.

Reviewer 4: ```suggestion
Alejandra González mentioned that getting changes into clippy is now bottlenecked by reviewers. Reviewing rustfmt code is complex. There's a lot of things to check, and edge cases to investigate. But Alejandra stressed that if someone is interested, they'd be happy to teach them, and they'd appreciate the help.
```

### Recoverable integer overflow

> Jana Dönszelmann opened a (draft) [pull request providing an numerical overflow handler](https://github.com/rust-lang/rust/pull/157314). This is something the team wanted for a long time. Currently, Rust will panic on integer overflows (e.g. adding `1` to a `u8` value of `255` — since there's no `256` in `u8`) in debug mode, and wrap (without panicking) in release.

Reviewer 4: ```suggestion
Jana Dönszelmann opened a (draft) [pull request providing a numerical overflow handler](https://github.com/rust-lang/rust/pull/157314). This is something the team has wanted for a long time. Currently, Rust will panic on integer overflows (e.g. adding `1` to a `u8` value of `255` — since there's no `256` in `u8`) in debug mode, and wrap (without panicking) in release.
```

The kernel needs the ability to detect overflows and react to them in some fashion. Jana's PR provides such a hook:

```rust
//@ no-prefer-dynamic
//@ compile-flags: -Coverflow-checks=recoverable
//@ run-pass
//@ check-run-results
#![feature(recoverable_integer_overflow)]
#![allow(arithmetic_overflow, unused)]

#[core::panic::integer_overflow_action]
fn overflow() {
    println!("overflow happened")
}

fn main() {
    let mut x = 255u8;
    x += 1;
}
```

The likely use case for this on the Linux side will be to print a kernel message the first time such a call happens (possibly tainting the kernel in the process) and keep going without spamming the log with more messages.

## Tomáš's July/August absence

> As a heads-up, Tomáš will be on a medical leave from 2026-07-12 until 2026-08-09. There is nothing to be worried about! Nurzhan will handle the work side of things.

Reviewer 4: ```suggestion
As a heads-up, Tomáš will be on medical leave from 2026-07-12 until 2026-08-09. There is nothing to be worried about! Nurzhan will handle the work side of things.
```

> We put our planned time-off in the [PM calendar](https://calendar.google.com/calendar/u/0?cid=N2RmMTJiZjc2YTFlZjU5MWNlYTQ4MmJjNDMxODEzYzcxZDg1MDViMTFhODk1NjQ1MzUxNjQ5ZTkwZWQ2NzUwNUBncm91cC5jYWxlbmRhci5nb29nbGUuY29t) and the [PM time off](https://rust-lang.zulipchat.com/#narrow/channel/392734-council/topic/PM.20time.20off/with/595113250) Zulip thread.

Reviewer 4: ```suggestion
We put our planned time off in the [PM calendar](https://calendar.google.com/calendar/u/0?cid=N2RmMTJiZjc2YTFlZjU5MWNlYTQ4MmJjNDMxODEzYzcxZDg1MDViMTFhODk1NjQ1MzUxNjQ5ZTkwZWQ2NzUwNUBncm91cC5jYWxlbmRhci5nb29nbGUuY29t) and the [PM time-off Zulip thread](https://rust-lang.zulipchat.com/#narrow/channel/392734-council/topic/PM.20time.20off/with/595113250).
```
